### PR TITLE
Copy Season Button on PR Leagues

### PR DESCRIPTION
**Pull request by JustinR17**
PR adds changes to the P/R overview page to include a 'copy season' button. This will copy the template, divisions and players from the selected season into a new season.

This also changes the season remove/copy buttons to use unique IDs and adds listener to the class field instead. (Addresses bug with only first season in list being deletable)

P/R Overview Page:
image

Copy Modal:
image

### CODE REVIEW

File: wlct/templates/pr.html [hunk 1]
```diff
@@ -39,6 +39,29 @@ <h4>{{ tournament.name }} &nbsp;
                 {% endif %}
 
     {% if tournament.created_by.token == request.session.token and not tournament.has_started %}
```
Comment (brendanfly) [inline wlct/templates/pr.html:41]: Does this mean you cannot copy a season that has started?

File: wlct/tournaments.py [hunk 2]
```diff
@@ -3919,6 +3919,47 @@ def create_season(self, season_name, games_at_once):
...
+        orig_season = PromotionalRelegationLeagueSeason.objects.filter(id=int(copy_season_id))
+        # Ensure parameters are correct before copying
+        if  not orig_season:
+            raise ValueError("Seasonal with id {} could not be found".format(copy_season_id))
+        # Get the # of games and template from original season
+        games_at_once = orig_season[0].games_at_once
+        template = orig_season[0].season_template;
+        if len(season_name) < 3 or len(season_name) > 251:
+            raise ValueError("Season name must be between 3-250 characters.")
+        if games_at_once < 1 or games_at_once > 100:
```
Comment (brendanfly) [inline wlct/tournaments.py:3933]: No need, the season you're copying from has to have a valid value.

File: wlct/tournaments.py [hunk 3]
```diff
@@ -3919,6 +3919,47 @@ def create_season(self, season_name, games_at_once):
...
+            raise ValueError("Games at once must be 1-100 inclusive.")
+        # Create a new season given original season
+        season = PromotionalRelegationLeagueSeason(pr_tournament=self, name=season_name, created_by=self.created_by, private=True, games_at_once=games_at_once, season_template=template)
```
Comment (brendanfly) [inline wlct/tournaments.py:3936]: Can use the pk trick here by setting orig_season.pk = None, orig_season.name = season_name, then save().

File: wlct/tournaments.py [hunk 4]
```diff
@@ -3919,6 +3919,47 @@ def create_season(self, season_name, games_at_once):
...
+        season.save()
+        
+        # Iterate through original season divisions and copy to new season
+        divisions = ClanLeagueDivision.objects.filter(pr_season=orig_season[0])
+        for div in divisions:
+            new_division = ClanLeagueDivision(title=div.title, pr_season=season)
+            new_division.save()
```
Comment (brendanfly) [inline wlct/tournaments.py:3943]: same here

File: wlct/templates/pr.html (hunk 1 above)
Comment (JustinR17) [reply wlct/templates/pr.html]: Good point, moved into new if block to check for created_by token only.

File: wlct/tournaments.py (hunk 2 above)
Comment (JustinR17) [reply wlct/tournaments.py]: Removed the checks

**Pull request merged**